src/frontend/src/optimizer/plan_node/logical_project.rs

```diff
@@ -132,8 +132,9 @@ impl LogicalProject {
         let i2o = Self::i2o_col_mapping_inner(input_schema.len(), exprs);
         input_pk
             .iter()
-            .filter_map(|pk_col| i2o.try_map(*pk_col))
-            .collect::<Vec<_>>()
+            .map(|pk_col| i2o.try_map(*pk_col))
+            .collect::<Option<Vec<_>>>()
+            .unwrap_or_default()
```

Comment:
.expect("pk lost during optimization") instead of returning an empty pk set?

Reply:
the batch query does not need the pk... I am thinking if we can move the pk property only in stream plan node.